### PR TITLE
docs: ALTER TABLE COMPACT

### PR DESCRIPTION
**Pull request by breezewish**
Signed-off-by: Wish <breezewish@outlook.com>

### What is changed, added or deleted? (Required)

Add docs for the new `ALTER TABLE ... COMPACT` syntax.

TODO: Update Release Note for this new feature.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): 

  - This new syntax is introduced by https://github.com/pingcap/tidb/pull/34741

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by shichun-0415 — APPROVED** [review]
Great job! Thanks.

File: sql-statements/sql-statement-alter-table-compact.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+你可以修改 TiFlash 配置文件参数 [`manual_compact_pool_size`](/tiflash/tiflash-configuration.md)，以更大资源占用为代价、获得更大的表级别并发度。例如，参数指定为 2 时，可以同时对 2 张表进行数据整理。
+
+## MySQL 兼容性
+
+`ALTER TABLE ... COMPACT` 语法是 TiDB 引入的对标准 SQL 语法的扩展。尽管没有对应的 MySQL 语法，但你仍然可通过 MySQL 5.7 ~ 8.0 各版本客户端，或各个遵循 MySQL 协议的数据库驱动执行该语句。
+
+## TiDB Binlog 及 TiCDC 兼容性
+
+`ALTER TABLE ... COMPACT` 语句不会导致逻辑数据变化，因而不会被 TiDB Binlog 及 TiCDC 同步到下游。
```
[the remark this answers is not in the document]
Comment (breezewish) [reply sql-statements/sql-statement-alter-table-compact.md:71]: 其他 SQL 语句会被默认同步，但这个语句不会被同步，还不太一样~

File: sql-statements/sql-statement-alter-table-compact.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+# ALTER TABLE ... COMPACT
+
+> **警告：**
+>
+> 该语句目前是实验性功能，不建议在生产环境中使用。
+
+发生写入后，TiDB 后台会自动进行数据整理（Compaction）。数据整理时，表中的物理数据会被重写，如清理已删除的数据、合并多版本数据等，从而可以获得更高的访问性能，并减少磁盘空间占用。使用 `ALTER TABLE ... COMPACT` 语句可以立即对指定的表进行数据整理，而无需等待后台触发。
+
+该语句执行时不会阻塞现有 SQL 语句的执行或使用 TiDB 功能，包括事务、DDL、GC 等，也不会改变通过 SQL 语句访问获得的数据内容。但执行过程中会占用一定量的 IO 及 CPU 资源，可能对业务延迟造成影响。
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-table-compact.md:14]:
```suggestion
该语句执行时不会阻塞现有 SQL 语句的执行或 TiDB 功能的使用，包括事务、DDL、GC 等，也不会改变通过 SQL 语句访问获得的数据内容。但执行过程中会占用一定量的 IO 及 CPU 资源，可能对业务延迟造成影响。
```

File: sql-statements/sql-statement-alter-table-compact.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+该语句会等待表中所有副本都数据整理完毕后才结束运行并返回。在执行过程中，你可以通过 [`KILL`](/sql-statements/sql-statement-kill.md) 语句安全地中断本张表的数据整理过程。中断不会破坏数据一致性或丢失数据，也不会影响后续重新发起或自动触发后台数据整理。
+
+目前仅支持对 TiFlash 进行数据整理，不支持对 TiKV 进行数据整理。
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-table-compact.md:18]:
```suggestion
目前该语句仅支持对 TiFlash 进行数据整理，不支持对 TiKV 进行数据整理。
```

File: sql-statements/sql-statement-alter-table-compact.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+```sql
+ALTER TABLE employee COMPACT TIFLASH REPLICA;
+```
+
+## 并发度
+
+`ALTER TABLE ... COMPACT` 语句会同时对表中所有副本发起数据整理操作。
+
+为了避免数据整理操作对在线业务造成显著影响，每个 TiFlash 实例默认只会同时处理一张表的数据整理操作（后台自动触发的数据整理不受影响）。这意味着，若你同时对多张表执行 `ALTER TABLE ... COMPACT` 语句，则它们在同一个 TiFlash 实例上会排队依次执行，而非同时得到执行。
```
Comment (qiancai) [inline sql-statements/sql-statement-alter-table-compact.md:61]:
```suggestion
为了避免数据整理操作对在线业务造成显著影响，在同一时间，每个 TiFlash 实例默认只会处理一张表的数据整理操作（后台自动触发的数据整理不受影响）。这意味着，若你同时对多张表执行 `ALTER TABLE ... COMPACT` 语句，则它们在同一个 TiFlash 实例上会排队依次执行，而非同时执行。
```

File: sql-statements/sql-statement-alter-table-compact.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+`ALTER TABLE ... COMPACT` 语法是 TiDB 引入的对标准 SQL 语法的扩展。尽管没有对应的 MySQL 语法，但你仍然可通过 MySQL 5.7 ~ 8.0 各版本客户端，或各个遵循 MySQL 协议的数据库驱动执行该语句。
+
+## TiDB Binlog 及 TiCDC 兼容性
+
+`ALTER TABLE ... COMPACT` 语句不会导致逻辑数据变化，因而不会被 TiDB Binlog 及 TiCDC 同步到下游。
+
+## 另请参阅
+
+- [ALTER TABLE](/sql-statements/sql-statement-alter-table.md)
+- [KILL TIDB](/sql-statements/sql-statement-kill.md)
```
Comment (breezewish) [inline sql-statements/sql-statement-alter-table-compact.md:76]:
```suggestion
- [KILL](/sql-statements/sql-statement-kill.md)
```

File: sql-statements/sql-statement-alter-table-compact.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+目前该语句仅支持对 TiFlash 进行数据整理，不支持对 TiKV 进行数据整理。
+
+## 语法图
+
+```ebnf+diagram
+AlterTableCompactStmt ::=
+    'ALTER' 'TABLE' TableName 'COMPACT' 'TIFLASH' 'REPLICA'
+```
+
+## 示例
```
Comment (moyun) [inline sql-statements/sql-statement-alter-table-compact.md:27]: 建议加一下缺省engine type时候的例子

**Review by moyun — COMMENTED** [review]
LGTM

**Pull request merged**